src/IO/H5/File.hpp

```diff
@@ -178,6 +178,7 @@ class H5File {
   hid_t file_id_{-1};
   mutable std::unique_ptr<h5::Object> current_object_{nullptr};
   std::vector<std::string> h5_groups_;
+  mutable bool throwing_exception_{false};
```

Comment:
Please add a code comment here what the purpose of this bool is and how the implementation handles opening/closing files in the context of exceptions.

Q: Is the bool set back to `false` at some point? Is it possible that the program recovers from an exception and then the state of this bool could be wrong, preventing the file from closing?

Reply:
You can use `std::uncaught_exceptions` to also handle errors from other parts of the code.

I think you could even throw if it's zero (or technically the same value as when the H5File object was constructed).  Everyone seems to say doing this is bad, but no one seems to give a coherent reason *why* it is bad, so I'm tempted to ignore them.